docs/porting-to-ROS2.md

```diff
@@ -154,6 +154,89 @@ A `tf2_ros::Buffer` member that is filled by a `tf2_ros::TransformListener` can
 
 
 
+### Dynamic reconfigure
+In ros2, the dynamic reconfigure feature is implemented by a parameter callback using the service. To use this feature, in the header file, you need to set a parameter handler and callback function as below.
```

Comment:
I would add a general restriction: what you describe is the use case where you really want to be able to change the value at any point during runtime and have your system respond to it.

Another, simpler use case, is to be able to set parameter values through a param file via the launch system. Then you take either the default value or what the user provided. But any later parameter change is ignored. This means your code is much much simpler, no callbacks etc. 

And from what I remember in our last daily, this latter use case is the only one we need to provide now.

Reply:
@fred-apex-ai 

> And from what I remember in our last daily, this latter use case is the only one we need to provide now

We need to port the features in dynamic reconfigure, and it is used in some nodes, for example in the mpc_follower.
The former use case is maybe about the discussion about the global parameter. 
The latter use case is for the dynamic reconfigure porting.

Reply:
Yes, even if we wouldn't need dynamic_reconfigure at all, it's probably helpful to have this information just as a general reference. But what (I think) Fred is saying makes sense, we could have a sentence at the start of this section that describes what the "dynamic reconfigure feature" is.

Reply:
Do you mean, we should have the separated section for the static parameter and dynamic parameter?
(Sorry, maybe I misunderstood due to the language problem. Correction of my English is welcome.)

Reply:
> And from what I remember in our last daily, this latter use case is the only one we need to provide now.

This is true **after we finish parameter tuning**. We still want to have this function when we tune to create a launch file. It is quite time consuming to relaunch nodes every time we change a value especially we have many parameters to tune.
I don't think this is mandatory, but is a good feature to have. If we can find a simple way to replace it in ROS2, I think it's worth doing it.